src/main/java/net/greghaines/jesque/worker/WorkerImpl.java

```diff
@@ -507,12 +507,14 @@ protected String getNextQueue() throws InterruptedException {
      */
     protected String pop(final String curQueue) {
         final String key = key(QUEUE, curQueue);
+        String inFlightKey = key(INFLIGHT, this.name, curQueue);
+
         switch (nextQueueStrategy) {
         case DRAIN_WHILE_MESSAGES_EXISTS:
-            return (String) this.jedis.evalsha(this.popScriptHash.get(), 3, key, key(INFLIGHT, this.name, curQueue),
+            return (String) this.jedis.evalsha(this.popScriptHash.get(), 3, key, inFlightKey,
                     JesqueUtils.createRecurringHashKey(key), Long.toString(System.currentTimeMillis()));
         case RESET_TO_HIGHEST_PRIORITY:
-            return (String) this.jedis.evalsha(this.multiPriorityQueuesScriptHash.get(), 1, curQueue);
+            return (String) this.jedis.evalsha(this.multiPriorityQueuesScriptHash.get(), 2, curQueue, inFlightKey, Long.toString(System.currentTimeMillis()));
```

Comment:
Or just a variable in the function if no other used in the class. I'm assuming no real difference in behavior if the millisecond is set a few microseconds before the call